distribution/pull_v2.go

```diff
@@ -392,9 +392,13 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named, platform
 		if p.config.RequireSchema2 {
 			return false, fmt.Errorf("invalid manifest: not schema2")
 		}
-		msg := schema1DeprecationMessage(ref)
-		logrus.Warn(msg)
-		progress.Message(p.config.ProgressOutput, "", msg)
+
+		// only warn on Docker Hub, to incite upgrading
+		if reference.Domain(ref) == "docker.io" {
```

Comment:
I wonder if we could/should make this more extensible... like have DockerHub (and those registries that want to do so) return a header saying "Please upgrade!". For our existing customers, we wouldn't return the header, but for *new* installations, we might do so (or if a customer explicitly decides to toggle it on)

Reply:
I like the idea of being able to have this sort of extra metadata come down from the registry. Deciding on such a protocol is a longer discussion though. Maybe propose this in another issue or even on the distribution spec.